src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,94 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final int VALID_AGE = 18;
+    private static final String VALID_PASSWORD = "password";
+    private static final String VALID_LOGIN = "admin admin";
+    private RegistrationService registrationService = new RegistrationServiceImpl();
+
+    @Test
+    void register_nullUser_notOk() {
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(null));
+    }
+
+    @Test
+    void register_passwordNull_notOk() {
+        User userWithNullPassword = new User();
+        userWithNullPassword.setPassword(null);
+        userWithNullPassword.setAge(VALID_AGE);
+        userWithNullPassword.setLogin(VALID_LOGIN);
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(userWithNullPassword));
+    }
+
+    @Test
+    void register_loginNull_notOk() {
+        User userWithNullLogin = new User();
+        userWithNullLogin.setPassword(VALID_PASSWORD);
+        userWithNullLogin.setAge(VALID_AGE);
+        userWithNullLogin.setLogin(null);
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(userWithNullLogin));
+    }
+
+    @Test
+    void register_ageNull_notOk() {
+        User userWithNullAge = new User();
+        userWithNullAge.setPassword(VALID_PASSWORD);
+        userWithNullAge.setAge(null);
+        userWithNullAge.setLogin(VALID_LOGIN);
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(userWithNullAge));
+    }
+
+    @Test
+    void register_invalidAge_notOk() {
+        User userYoungerThen18 = new User();
```

Comment:
Minor comment - The name of the test method should clearly describe what you are going to test, so you don't need to name the user userYoungerThen18, just user, because sometimes you will test several parameters and the name of object will be too big ))